tests/unittest/_torch/visual_gen/test_offloading.py

```
@@ -0,0 +1,443 @@
+        }
+    )
+
+    assert args.offload_config.stages == ["future_model_stage"]
+
+
+def test_pipeline_rejects_unknown_offload_stage_names_for_this_model():
+    pipeline = _CustomOffloadPipeline(_make_config(offload_stages=["transformer.blocks"]))
+
+    with pytest.raises(
+        ValueError,
+        match="Unknown offload_config.stages entries.*transformer\\.blocks.*denoising_transformer",
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)